src/AmalgamatorSession.ts

```diff
@@ -401,10 +405,16 @@ export class AmalgamatorSession extends LoggingDebugSession {
         const childResponse = await childDap.stackTraceRequest(args);
         const frames = childResponse.body.stackFrames;
         // XXX: When does frameHandles get reset as we don't have a "stopped all"
-        frames.forEach(
-            (frame) =>
-                (frame.id = this.frameHandles.create([childDap, frame.id]))
-        );
+        frames.forEach((frame) => {
+            frame.id = this.frameHandles.create([childDap, frame.id]);
+            if (frame.instructionPointerReference) {
+                this.addressMap.set(childIndex, [
+                    frame.instructionPointerReference,
+                    this.addressMap.get(childIndex)?.[1] as string,
+                    this.addressMap.get(childIndex)?.[2] as string,
+                ]);
```

Comment:
I think this map should be setup the other way, i.e. a map of addresses -> child dap index.

That would work because you have an assumption (which looks correct, but undocumented) that the only addresses coming from the VSCode will be the start/end addresses or the instructionPointerReference

That would make this code something like this:

```typescript
this.addressMap.set(frame.instructionPointerReference,childIndex);
```

It also would naturally be a fifo, allowing a later update for the same address from a different DAP to take precedence. The current solution will always prefer whichever entry iterates first, even though it may be older.

Reply:
I have updated it. Your suggestion is very helpful to me. Thank so much.